returnn/forward.py

```diff
@@ -0,0 +1,213 @@
+        mem_rqmt=4,
+        cpu_rqmt=4,
+        returnn_python_exe=None,
+        returnn_root=None,
+    ):
+        """
+
+        :param Checkpoint model_checkpoint: Checkpoint object pointing to a stored RETURNN Tensorflow model
+        :param ReturnnConfig returnn_config: RETURNN config dict
+        :param dict returnn_post_config: RETURNN config dict (no hashing)
+        :param list[str] hdf_outputs: list of additional hdf output layer file names that the network generates (e.g. attention.hdf);
+          The hdf outputs have to be a valid subset or be equal to the hdf_dump_layers in the config.
+        :param bool eval_mode: run forward in eval mode, the default hdf is not available in this case and no search will be done.
+        :param int log_verbosity: RETURNN log verbosity
+        :param str device: RETURNN device, cpu or gpu
+        :param int time_rqmt: job time requirement
+        :param int mem_rqmt: job memory requirement
+        :param int cpu_rqmt: job cpu requirement
+        :param Path|str returnn_python_exe: path to the RETURNN executable (python binary or launch script)
+        :param Path|str returnn_root: path to the RETURNN src folder
+        """
+        self.returnn_python_exe = (
+            returnn_python_exe
+            if returnn_python_exe is not None
+            else gs.RETURNN_PYTHON_EXE
+        )
+        self.returnn_root = (
+            returnn_root if returnn_root is not None else gs.RETURNN_ROOT
+        )
+
+        self._model_checkpoint = model_checkpoint
+        self._returnn_config = returnn_config
+        self._eval_mode = eval_mode
+        self._log_verbosity = log_verbosity
+        self._device = device
+
+        self.out_returnn_config_file = self.output_path("returnn.config")
+
+        self.out_hdf_files = {}
+        hdf_outputs = hdf_outputs if hdf_outputs else []
+        for output in hdf_outputs:
+            self.out_hdf_files[output] = self.output_path(output)
+        if not eval_mode:
+            self.out_hdf_files["output.hdf"] = self.output_path("output.hdf")
+            self.out_default_hdf = self.out_hdf_files["output.hdf"]
+
+        self.rqmt = {
+            "gpu": 1 if device == "gpu" else 0,
+            "cpu": cpu_rqmt,
+            "mem": mem_rqmt,
+            "time": time_rqmt,
+        }
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", resume="run", rqmt=self.rqmt)
+
+    def create_files(self):
+        config = self.create_returnn_config(
+            model_checkpoint=self._model_checkpoint,
+            returnn_config=self._returnn_config,
+            eval_mode=self._eval_mode,
+            log_verbosity=self._log_verbosity,
+            device=self._device,
+        )
+        config.write(self.out_returnn_config_file.get_path())
+
+        cmd = [
+            tk.uncached_path(self.returnn_python_exe),
+            os.path.join(tk.uncached_path(self.returnn_root), "rnn.py"),
```

Comment:
avoid `tk.uncached_path`

Reply:
If we do this, then should we drop `gs.` support completely?

Reply:
"should" probably yes. "can" probably no.. see the whole `gs.RETURNN_ROOT` discussion.

There are some sisyphus settings that I think are fine to read, e.g. temp prefix for local caching - as long as there is no implementation for this in sisyphus directly.